Cabal/src/Distribution/PackageDescription/Check/Warning.hs

```diff
@@ -25,7 +25,7 @@ module Distribution.PackageDescription.Check.Warning
   , ppPackageCheck
   , ppCheckExplanationId
   , isHackageDistError
-  , extractCheckExplantion
+  , extractCheckExplanation
```

Comment:
Given that we export pretty much everything, this counts as an API change.

Reply:
If we don't backport it, why is it a problem? Changing API is what major versions are for.

@philderbeast I don't feel strong about it, so perhaps, pragmatically speaking, it'd be easier merge without this change.

Reply:
I checked hoogle for it and couldn't see it, checked hackage/Cabal and couldn't see it and checked the source. It is only used in `Distribution.PackageDescription.Check.Monad` and not re-exported isn't it?

```
$ grep -R -E 'Explantion' ./**/*.hs
./Cabal/src/Distribution/PackageDescription/Check/Monad.hs:    isErrAllowable c = case extractCheckExplantion c of
./Cabal/src/Distribution/PackageDescription/Check/Warning.hs:  , extractCheckExplantion
./Cabal/src/Distribution/PackageDescription/Check/Warning.hs:        . extractCheckExplantion
./Cabal/src/Distribution/PackageDescription/Check/Warning.hs:extractCheckExplantion :: PackageCheck -> CheckExplanation
./Cabal/src/Distribution/PackageDescription/Check/Warning.hs:extractCheckExplantion (PackageBuildImpossible e) = e
./Cabal/src/Distribution/PackageDescription/Check/Warning.hs:extractCheckExplantion (PackageBuildWarning e) = e
./Cabal/src/Distribution/PackageDescription/Check/Warning.hs:extractCheckExplantion (PackageDistSuspicious e) = e
./Cabal/src/Distribution/PackageDescription/Check/Warning.hs:extractCheckExplantion (PackageDistSuspiciousWarn e) = e
./Cabal/src/Distribution/PackageDescription/Check/Warning.hs:extractCheckExplantion (PackageDistInexcusable e) = e
```

Reply:
It's in `exposed-modules`. Have you checked all of Hackage for it?

Reply:
As to the import of an API change here:

- at the very least, it needs a changelog;
- this may be a good test case for @Kleidukos's deprecation strategy, especially since I suspect that if someone _were_ using it, they would probably have mentioned the misspelling to us (I hope).

Reply:
The above said, I'm not sure a test case is very useful considering that we may well have an actual need for it well before the "test" deprecation cycle ended. It's a difficulty with long-running protocols.

Reply:
Hm, I thought I dropped the start of this on an export list entry? It isn't in the API files I just generated or the ones from `master`

Reply:
No, seems it's in `other-modules` after all. The API check is correct, and you're clear.

Reply:
Are you able to approve this PR @geekosaur?

Reply:
Yes, I just wasn't awake yet.